### ISSUE TITLE
$fish_color_search_match only supports background colors

### ISSUE THREAD

**Issue by alphapapa** [fish-shell/fish-shell#2442]
It seems that `fish_color_search_match` only supports background colors.  Doing things like:
```fish
set fish_color_search_match --underline
set fish_color_search_match yellow --underline --background=black
set fish_color_search_match yellow
```
...seem to have no effect.  But this works:
```fish
set fish_color_search_match --background=black
```

**Comment by netmute** [comment]
Just wanted to put this on the radar again.

**Comment by cmansley** [comment]
Is this request to expand the support for the [set_color](https://fishshell.com/docs/current/commands.html#set_color) spec in `fish_color_search_match` except for the foreground color? 

Because currently, there are (at least) three colors that describe the foreground text color, `fish_pager_color_prefix`, `fish_pager_color_completion` and `fish_pager_color_description`. The options would be to override all three colors with the foreground color given in `fish_color_search_match` or support everything except the foreground color in the `fish_color_search_match`.

The original request from @rafaelrinaldi in #2593 mentioned changing the text color 

> So, the text color is not customizable at all?

But, I don't think it would be possible without breaking the `fish_color_search_match` out into three different variables for the selected versions of each of the `fish_pager_*` variables. Like `fish_pager_color_prefix_selected`. Otherwise, I think it would be conceptually confusing. 

I honestly don't see why adding italics (or anything else) to the `fish_color_search_match` already doesn't work correctly and italicize the whole matched line. It seems like it is already an "or" of the background and foregrounds in the [set_color](https://github.com/fish-shell/fish-shell/blob/3b3047c3f6cf10d6e4c076d44ca0949864a93e03/src/output.cpp#L167) function. The fact that the font modifiers don't work seems like the bug. But, explicitly excluding the foreground color seems like a feature in favor of simplicity.

**Comment by alphapapa** [comment]
> Is this request to expand the support for the set_color spec in fish_color_search_match except for the foreground color?

I had completely forgotten about this.  But as I understand it (i.e. what I was thinking 2 years ago), the request is to support all color specs in `fish_color_search_match`.  Why exclude foreground color?  For example, it would be nice to be able to set `fish_color_search_match` to a slightly brighter color than the default.

**Comment by cmansley** [comment]
Yep. I agree changing the font modifiers would be awesome (dim, bold, italics, etc...). But, changing the font color may conflict with the three other variables I listed, because they also effect the color of the font. (logically conflict, we can do anything we want :)

Here are two gifs. The first is the current behavior.
![peek 2017-12-27 10-41](https://user-images.githubusercontent.com/97713/34390352-28e0437c-eaf4-11e7-9891-0cb6d0efad51.gif)
The second is the complete overriding of the selected foreground and background color with `fish_color_search_match`. However, this doesn't allow for us to set the background only, leaving the colors from `fish_pager_color_*` variables.
![peek 2017-12-27 10-50](https://user-images.githubusercontent.com/97713/34390384-5ce411f8-eaf4-11e7-872f-949c320bd72e.gif)

**Comment by alphapapa** [comment]
Thanks for making the GIFs, that illustrates it very well!

**Comment by cmansley** [comment]
Ok so the question still remains. Does the second gif represent expected behavior? That the style of the selected object totally overwhelms the style selected in the `fish_pager_color_*` variables. Or do we want to complicate the logic to enable the current behavior and some parts of the second gif behavior?

**Comment by alphapapa** [comment]
Good question.  I'm not sure.  I guess, if anyone decides to work on it, they can decide.  :)

**Comment by max-sixty** [comment]
Do the included color schemes handle this correctly?

Currently ayu Light has:

```
# ayu Light in fish_config

set -U fish_color_selection FF9940
set -U fish_color_search_match FF9940
```

But this doesn't to be visible, here `testing` is selected:

<img width="163" alt="image" src="https://user-images.githubusercontent.com/5635139/116626051-810a6100-a8ff-11eb-9a59-04cb7dcb2c9d.png">


vs. after setting

```
set fish_color_search_match --background=FF9940
set fish_color_selection --background=FF9940
```

<img width="162" alt="image" src="https://user-images.githubusercontent.com/5635139/116626099-92536d80-a8ff-11eb-9aa0-77da5ea2bbc3.png">
